ehsm_kms_service/enroll_app/main.cpp

```diff
@@ -0,0 +1,308 @@
+        0x85, 0xd0, 0x3a, 0x62, 0x37, 0x30, 0xae, 0xad,
+        0x3e, 0x3d, 0xaa, 0xee, 0x9c, 0x60, 0x73, 0x1d,
+        0xb0, 0x5b, 0xe8, 0x62, 0x1c, 0x4b, 0xeb, 0x38
+    },
+    {
+        0xd4, 0x81, 0x40, 0xd9, 0x50, 0xe2, 0x57, 0x7b,
+        0x26, 0xee, 0xb7, 0x41, 0xe7, 0xc6, 0x14, 0xe2,
+        0x24, 0xb7, 0xbd, 0xc9, 0x03, 0xf2, 0x9a, 0x28,
+        0xa8, 0x3c, 0xc8, 0x10, 0x11, 0x14, 0x5e, 0x06
+    }
+};
+
+string base64_decode(const string &encoded_string) {
+    uint32_t in_len = encoded_string.size();
+    uint32_t i = 0;
+    uint32_t j = 0;
+    uint32_t in_ = 0;
+    unsigned char char_array_4[4], char_array_3[3];
+    string decode_str;
+
+    while (in_len-- && ( encoded_string[in_] != '=') && is_base64(encoded_string[in_])) {
+        char_array_4[i++] = encoded_string[in_]; in_++;
+        if (i ==4) {
+            for (i = 0; i <4; i++)
+                char_array_4[i] = static_cast<unsigned char>(encode_table.find(char_array_4[i]));
+
+            char_array_3[0] = (char_array_4[0] << 2) + ((char_array_4[1] & 0x30) >> 4);
+            char_array_3[1] = ((char_array_4[1] & 0xf) << 4) + ((char_array_4[2] & 0x3c) >> 2);
+            char_array_3[2] = ((char_array_4[2] & 0x3) << 6) + char_array_4[3];
+
+            for (i = 0; (i < 3); i++)
+                decode_str += char_array_3[i];
+            i = 0;
+        }
+    }
+
+    if (i) {
+        for (j = i; j <4; j++)
+            char_array_4[j] = 0;
+
+        for (j = 0; j <4; j++)
+            char_array_4[j] = static_cast<unsigned char>(encode_table.find(char_array_4[j]));
+
+        char_array_3[0] = (char_array_4[0] << 2) + ((char_array_4[1] & 0x30) >> 4);
+        char_array_3[1] = ((char_array_4[1] & 0xf) << 4) + ((char_array_4[2] & 0x3c) >> 2);
+        char_array_3[2] = ((char_array_4[2] & 0x3) << 6) + char_array_4[3];
+
+        for (j = 0; (j < i - 1); j++)
+            decode_str += char_array_3[j];
+    }
+
+    return decode_str;
+}
+
+static char* StringToChar(string str)
+{
+    char* returnChar = nullptr;
+    if (str.size() > 0) {
+        int len = str.size() + 1;
+        returnChar = (char *)malloc(len * sizeof(uint8_t));
+        if(returnChar != nullptr){
+            memset(returnChar, 0, len);
+            memcpy(returnChar, str.c_str(), len);
+        }
+    }
+    return returnChar;
+}
+
+
+Json::Value readConfigJson()
+{
+    Json::Reader reader;
+	Json::Value kmsConfigJson;
+    string ehsm_kms_ip;
+
+    ifstream inFile("config.json", ios::binary);
+    if (!inFile.is_open()){
+        cout << "Error: opening file"<< endl;
+        return kmsConfigJson;
+    }
+	reader.parse(inFile, kmsConfigJson);
+    inFile.close();
+	return kmsConfigJson;
+}
+
+
+// curl callback func
+size_t req_callback(void *ptr, size_t size, size_t nmem, void *stream)
+{
+	string *str = (string*)stream;
+	(*str).append((char*)ptr, size*nmem);
+	return size * nmem;
+}
+
+string post(string url, string content){

```

Comment:
put the rest common functions into another file, please reference the base64.cpp/base64.h

Reply:
It will be fix